vidaud/seta
# [seta-react] Refactor routes and use react-router navigation

## Description

Extracted the application routes to a separate component and made all the links in the `Header` use proper `react-router` navigation, so that the app behaves like a SPA and doesn't do a full page reload for every link.

Also:
- Moved the app logo from the GitHub URL to the public folder
- Implemented a Layout component on the main route
- Made some minor adjustments to the header styling
- Defined a couple of utility types
- The current page link in the header now remains selected after clicking on it, and also on refresh

## How to test

- Make sure that all the links in the header navigate as expected and that the browser doesn't do a full page reload
- Verify that the current page link in the header stays selected after clicking on it and after refreshing the page

### Important

The links in the breadcrumbs that appear on some of the pages (eg. Search) still do a full reload - they will be fixed on a separate PR as there is some complexity to make this generic.
